src/send.c

```diff
@@ -489,3 +482,25 @@ int send_run(sock_t st, shard_t *s)
 	log_debug("send", "thread %hu cleanly finished", s->thread_id);
 	return EXIT_SUCCESS;
 }
+
+batch_t* create_packet_batch(uint8_t capacity) {
+	batch_t* batch = malloc(sizeof(batch_t));
```

Comment:
If you want to be real cool, allocate this all in a single malloc call and slice off chunks of it youself, such that all of the memory for batch and what it points to is contiguous. It should help with cache locality.

Reply:
Oooo, that's a great idea!